### ISSUE TITLE
Regularizers should have options to not regularize frozen parameters

### ISSUE THREAD

**Issue by codedecde** [allenai/allennlp#2790]
Currently, the regularizers have no way of differentiating between frozen and non frozen parameters. For example, consider the use case of building a tagger using BERT embeddings. To apply l2 loss on the entire model, a regex .*weight.* is enough. However, this also regularizes the bert layers, something that I'd like to avoid

**Describe the solution you'd like**
A boolean for the RegularizerApplicator like:
ignore_frozen_parameters
Which when true, can avoid parameters where requires_grad is False 

**Describe alternatives you've considered**
The alternative is specifying each layer individually, which is tedious.

**Comment by joelgrus** [comment]
if the penalty is just additive in the weights, then does it matter if it includes weights of frozen layers? we'll just never compute the gradients for those, right?

**Comment by codedecde** [comment]
It just messes up the loss logging.
For example, for any general classification loss, the range is usually around ~1 to < 1
But now, the loss would be ||weights||^{2} + loss, which can be quite large for huge models (for BERT, its order 10^3). So makes tracking model progress cumbersome.

**Comment by schmmd** [comment]
Sounds reasonable, we would welcome a contribution.
